app/validators/termlist_has_no_duplicates.rb

```diff
@@ -0,0 +1,21 @@
+# Verifies that all terms in a list are unique within the list
+class TermlistHasNoDuplicates < ActiveModel::Validator
+  def validate(record)
+    @uris_seen = Hash.new(0)
+    record.terms.each do |term|
```

Comment:
I guess it comes down to whether we want to have an indication of which item is duped.  With just a check on list length, all we can say is "your list has stuff it shouldn't".